sklearn/ensemble/_hist_gradient_boosting/tests/test_loss.py

```diff
@@ -211,11 +218,33 @@ def test_baseline_categorical_crossentropy():
         assert np.allclose(baseline_prediction[k, :], np.log(p))
 
 
+def test_baseline_poisson_loss():
+    rng = np.random.RandomState(0)
+
+    loss = _LOSSES['poisson_loss'](sample_weight=None)
+    y_train = rng.poisson(size=100).astype(np.float64)
+    # Make sure at least one sample point is larger than zero
+    y_train[0] = 1.
```

Comment:
```suggestion
    # Sanity check, make sure at least one sample is non-zero so we
    # don't take log(0)
    assert y_train.sum() > 0
```

Reply:
Though TBF I'm fine removing this given the very low probability that this ever happens (basically if it passes all the CIs, it's fine)